mount.js

```diff
@@ -53,7 +53,45 @@ export function mountJsx (selector, ComponentToMount, additionalProps = {})
         }
         catch (e)
         {
-            // ignore
+            console.error("Could not mount component", e);
+        }
+    }
+}
+
+
+/**
+ * Mounts the component as JSX Component.
+ * The matched node must contain the data as JSON encoded content.
+ * The content will be automatically encapsulated in a unistore store.
+ *
+ * @param {string} selector
+ * @param {preact.Component} ComponentToMount
+ * @param {Object} additionalProps
+ */
+export function mountJsxWithStore (selector, ComponentToMount, additionalProps = {})
+{
+    const elements = find(selector);
+
+    for (let i = 0; i < elements.length; i++)
+    {
+        try
+        {
+            const node = elements[i];
+            let data = JSON.parse(node.textContent);
+            data = merge(data, additionalProps);
+            let store = createStore(data);
+
+            render(
+                h(Provider, {store},
+                    h(ComponentToMount, data)
+                ),
```

Comment:
Changing the file extension could be a BC break, so we just keep nesting.

Also, as this is still very readable, porting it to JSX wouldn't change much.